rpcs3/Loader/TROPUSR.cpp

```diff
@@ -215,6 +216,46 @@ u32 TROPUSRLoader::GetUnlockedTrophiesCount()
 	return count;
 }
 
+u32 TROPUSRLoader::GetUnlockedPlatinumID(u32 trophy_id)
+{
+	constexpr u32 invalid_trophy_id = -1; // SCE_NP_TROPHY_INVALID_TROPHY_ID;
+
+	if (trophy_id >= m_table6.size() || trophy_id >= m_table4.size())
+	{
+		return invalid_trophy_id;
+	}
+
+	// Get this trophies platinum link id
+	const u32 pid = m_table4[trophy_id].trophy_pid;
+
+	// A platinum trophy has to have a valid trophy id
+	if (pid == invalid_trophy_id)
+	{
+		return invalid_trophy_id;
+	}
+
+	for (const auto& trophy : m_table6)
+	{
+		for (const auto& trophy_info : m_table4)
+		{
+			if (trophy_info.trophy_id == trophy.trophy_id)
+			{
+				// The platinum trophy stays locked if any relevant trophy is still locked
+				if (trophy_info.trophy_pid == pid && !trophy.trophy_state)
+				{
+					return invalid_trophy_id;
+				}
+
+				// Onto the next trophy id
+				break;
```

Comment:
Unless I'm misunderstanding the intent, this begs for a `std::find_if`,

Reply:
Yeah I was just too lazy because I had no idea if this would work anyway